cmd/csi-provisioner/csi-provisioner.go

```diff
@@ -638,12 +646,31 @@ func main() {
 		}
 
 		if capacityController != nil {
-			go capacityController.Run(ctx, int(*capacityThreads))
+			controllerWg.Add(1)
+			go func() {
+				defer controllerWg.Done()
+				capacityController.Run(ctx, int(*capacityThreads))
+				klog.Info("capacity controller finished shutdown")
+			}()
 		}
 		if csiClaimController != nil {
-			go csiClaimController.Run(ctx, int(*finalizerThreads))
+			controllerWg.Add(1)
+			go func() {
+				defer controllerWg.Done()
+				csiClaimController.Run(ctx, int(*finalizerThreads))
+				klog.Info("csi claim controller finished shutdown")
+			}()
 		}
-		provisionController.Run(ctx)
+
+		controllerWg.Add(1)
+		go func() {
+			defer controllerWg.Done()
+			provisionController.Run(ctx)
+			klog.Info("provisioning controller finished shutdown")
+		}()
+
+		controllerWg.Wait()
+		klog.Info("all controllers finished shutdown")
```

Comment:
I don't think any of the changes to `main()` matter because `provisionController.Run(ctx)` is blocking forever until the process exits.

Reply:
I think this is only for logging, can revert all of these changes if you dont want this. Just thought it would be proper hygiene.